lib/opal/nodes/top.rb

```diff
@@ -41,11 +49,19 @@ def compile
           line body_code
         end
 
+        prepend_import_lines if compiler.es6_modules?
+
         closing
       end
 
-      def opening
-        if compiler.requirable?
+      def opening(module_name = nil)
+        if compiler.es6_modules?
+          # to enable some webpack features for opal-webpack-loader it has to refer to the opal code from within webpack later on
+          # so we give it a handle 'opal_code'
+          line 'const opal_code = function() {'
+          # global makes sure we get the webpack global context and its Opal.modules, and not a locally shielded Opal.modules
+          line "  global.Opal.modules[#{module_name.inspect}] = function(Opal) {"
```

Comment:
Is `global` defined in a browser context? Maybe using globalThis would be a better idea? Maybe it doesn't matter, I don't know, it just caught my eye. See #2042

Reply:
This global here is from webpack. global has special meaning within webpack. Webpack usually gives modules their own context. Referring to global makes sure webpack provides the global context. So using `Opal.modules...` here would provide a local, incomplete version of `Opal`, using `global.Opal` makes sure the real Opal is used.

So within imported modules:
`Opal.modules != global.Opal.modules`

Think of `global` here as a compiler directive maybe.